pkg/services/validator/azure.go

```diff
@@ -60,19 +65,65 @@ func readAzurePluginRules(vc *components.ValidatorConfig, _ *cfg.TaskConfig, _ k
 }
 
 func readAzureCredentials(c *components.AzurePluginConfig, tc *cfg.TaskConfig, k8sClient kubernetes.Interface) error {
+	if tc.Direct {
+		return readDirectAzureCredentials(c)
+	}
+	return readInstallAzureCredentials(c, k8sClient)
+}
+
+func readDirectAzureCredentials(c *components.AzurePluginConfig) error {
+	api, err := azure.NewAzureAPI()
+	if err != nil {
+		return fmt.Errorf("failed to create Azure API: %w", err)
+	}
+	_, err = api.RoleAssignmentsClient.Get(context.Background(), "00000000-0000-0000-0000-000000000000", "00000000-000000-0000000000", nil)
+	// auth toolchain is configured, skip prompting for credentials
+	if err == nil || !strings.Contains(err.Error(), azureNoCredsErr) {
+		return nil
```

Comment:
Can't do an early return here without exporting env vars.

Reply:
This should be fine. For the early return to happen, we need either valid creds in the env, or elsewhere that the sdk can use. This is the equivalent of what we had before, where if valid credentials are configured on the system, direct mode will work without export. If I'm missing something please let me know!